mypy/checkstrformat.py

```diff
@@ -518,7 +546,21 @@ def apply_field_accessors(self, spec: ConversionSpecifier, repl: Expression,
 
         # This is a bit of a dirty trick, but it looks like this is the simplest way.
         temp_errors = self.msg.clean_copy().errors
-        dummy = DUMMY_FIELD_NAME + spec.field[len(spec.key):]
+
+        try:
+            first_obj, split_fields = string._string.formatter_field_name_split(
+                spec.field[len(spec.key):])
+            split_fields_str = [
+                ("." + str(field)) if is_attr else
+                '[{}]'.format(field) if isinstance(field, int) else
+                '["{}"]'.format(field.replace('"', '\\"')) for is_attr, field in split_fields]
+        except Exception:
```

Comment:
We avoid catching `Exception` since it can be triggered by unexpected error conditions. Can you make this more specific? Also, if you narrow this down to the actual call that can fail instead of several lines of code, it would be easier to see what is likely to raise an exception.

Reply:
I narrowed it down to a ValueError but either formatter_field_name_split or the iteration itself can throw it so I don't see much point separating them.